### DISCUSSION TITLE
First up/down arrow key press not switching items on static Combobox options

### DISCUSSION THREAD

**Discussion by oliwheeler** [tailwindlabs/headlessui#2961] [Help] [answered]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

1.7.18

**What browser are you using?**

Reproducible on Chrome, Safari, Firefox.

**Reproduction URL**

[Code sandbox Repro link](https://codesandbox.io/p/devbox/cool-feistel-k8cvcd?file=%2Fsrc%2FApp.jsx%3A11%2C22&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clri93n8k00073b6imtvbqns5%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clri93n8j00023b6ig8djbr20%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clri93n8j00043b6iggasfjji%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clri93n8k00063b6iohcq5uts%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clri93n8j00023b6ig8djbr20%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clri93n8j00013b6i7nld8ejr%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clri99ffq001z3b6js0cuioub%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A11%252C%2522startColumn%2522%253A22%252C%2522endLineNumber%2522%253A11%252C%2522endColumn%2522%253A22%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.jsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clri93n8j00023b6ig8djbr20%2522%252C%2522activeTabId%2522%253A%2522clri99ffq001z3b6js0cuioub%2522%257D%252C%2522clri93n8k00063b6iohcq5uts%2522%253A%257B%2522id%2522%253A%2522clri93n8k00063b6iohcq5uts%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clri93n8k00053b6iq63trb4t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clri93n8k00053b6iq63trb4t%2522%257D%252C%2522clri93n8j00043b6iggasfjji%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clri93n8j00033b6iuw98drn5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522clri93n8j00043b6iggasfjji%2522%252C%2522activeTabId%2522%253A%2522clri93n8j00033b6iuw98drn5%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Describe your issue**

Click the Panel Button to open panel containing the static combobox. An option is already pre-selected
Press up/down arrow keys to switch between options.
The first up/down arrow key press is does nothing and doesn't move between the options as expected.

**Comment by adamwathan** [comment, marked as answer]
Hey! So the important thing to understand is that the `static` prop is just for you to take control of whether the element is visible or not, but it doesn't influence the actual _open_ state of the comobobox. Even though you can see the panel, the combobox is technically closed from an internal state perspective. What's happening when you hit up/down the first time is that the combobox is now opening, and the second up/down actually changes the selected item.

The `static` prop exists for integrating the component with things like Framer Motion where you need to let Framer Motion handle whether something is shown or not using a component like `AnimatePresence` — it's not meant for permanently showing the combobox list and making it permanently interactive, just for giving control over hiding it to something else, but it's still meant to be hidden when the combobox is in a closed state and behaves like it's hidden.

I'm not totally sure the best way to build what you're trying to build with Headless UI but will have a think on it. Perhaps hiding the `Combobox.Input` unless the combobox is open, and using `Combobox.Button` to hide/show. I'd play with that and see what you are able to do.

Going to convert this to a discussion because this is expected behavior but happy to help riff on a way to build the UI you are trying to build 👍

**Comment by oliwheeler** [comment]
Thanks for the clear explanation!

I was trying to recreate this shadcn/ui Combobox component using only HeadlessUI components: https://ui.shadcn.com/docs/components/combobox

**Comment by tkapitein** [comment]
Hello @adamwathan, I previously thought of a similar solution as you've now presented. However, there's an issue. The Combobox.Button component automatically gets a tabIndex="-1" property which I was unable to override. Is there a way to ensure this property can be overridden?

**Comment by ghost** [comment]
Hey!

Any solutions on this case? I am trying to do the same.
